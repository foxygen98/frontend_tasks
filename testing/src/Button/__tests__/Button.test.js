import React from 'react';
import {render, cleanup, fireEvent, screen} from '@testing-library/react';
import Button from '../Button';

const onClick = jest.fn();

describe('<Button />', () => {
  afterEach(cleanup);

  it('should render button', () => {
    const wrapper = render(<Button buttonText="Button" />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should be clickable', () => {
    const wrapper = render(<Button buttonText="Button" onClick={onClick}/>);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('onClick should be called', () => {
    render(<Button buttonText="Button" onClick={onClick}/>);
    fireEvent.click(screen.getByRole('button'));
    expect(onClick).toHaveBeenCalled();
  });
})
