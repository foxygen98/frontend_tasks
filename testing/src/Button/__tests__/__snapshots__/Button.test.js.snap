// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should be clickable 1`] = `
<DocumentFragment>
  <button>
    Button
  </button>
</DocumentFragment>
`;

exports[`<Button /> should render button 1`] = `
<DocumentFragment>
  <button>
    Button
  </button>
</DocumentFragment>
`;
